src/canisters/frontend/ic-certified-assets/src/http.rs

```diff
@@ -80,6 +80,47 @@ impl HttpRequest {
             .iter()
             .find_map(|(k, v)| k.eq_ignore_ascii_case(header_key).then_some(v))
     }
+
+    pub fn redirect_from_raw_to_certified_domain(&self) -> HttpResponse {
+        #[cfg(not(test))]
+        let location = format!(
+            "https://{canister_id}.ic0.app{path}",
+            canister_id = ic_cdk::api::id().to_text(),
+            path = self.url
+        );
+        #[cfg(test)]
+        let location = format!(
+            "https://{canister_id}.ic0.app{path}",
+            canister_id = self.get_canister_id(),
+            path = self.url
+        );
```

Comment:
Is this necessary after updating the ic0 crate?

Reply:
looks like its still necessary. attempting to run `cargo test` while relying only on `ic_cdk::api::id()` results in the following error
```
thread 'tests::allow_raw_access::redirects_from_raw_to_certified' panicked at 'canister_self_size should only be called inside canisters.', ~/.cargo/registry/src/github.com-1ecc6299db9ec823/ic0-0.18.7/src/ic0.rs:118:9
```

Reply:
Okay, how about some DRY then

```suggestion
        #[cfg(not(test))]
        let canister_id = ic_cdk::api::id().to_text();
        #[cfg(test)]
        let canister_id = self.get_canister_id();

        let location = format!(
            "https://{canister_id}.ic0.app{path}",
            canister_id,
            path = self.url
        );
```